app/models/user.rb

```diff
@@ -26,6 +26,6 @@ def generate_token(column)
   end
 
   def works_here?(id)
-    @current_user.merchant.id == id
+    self.merchant.id == id
```

Comment:
i tested this pretty well but maybe someone just check me that this is right. its the user model so self should be the same as current user